core/commands/add.go

```diff
@@ -265,24 +272,73 @@ type adder struct {
 	progress bool
 	hidden   bool
 	trickle  bool
+	chunker  string
+}
+
+func getChunker(r io.Reader, chunker string) (chunk.Splitter, error) {
+	switch {
+	case chunker == "":
+		return chunk.NewSizeSplitter(r, chunk.DefaultBlockSize), nil
+	case strings.HasPrefix(chunker, "size-"):
+		sizeStr := strings.Split(chunker, "-")[1]
+		size, err := strconv.Atoi(sizeStr)
+		if err != nil {
+			return nil, err
+		}
+
+		return chunk.NewSizeSplitter(r, int64(size)), nil
+	case strings.HasPrefix(chunker, "rabin"):
+		parts := strings.Split(chunker, "-")
+		switch len(parts) {
+		case 1:
+			return chunk.NewRabin(r, uint64(chunk.DefaultBlockSize)), nil
+		case 2:
+			size, err := strconv.Atoi(parts[1])
+			if err != nil {
+				return nil, err
+			}
+			return chunk.NewRabin(r, uint64(size)), nil
+		case 4:
+			min, err := strconv.Atoi(parts[1])
+			if err != nil {
+				return nil, err
+			}
+			avg, err := strconv.Atoi(parts[2])
+			if err != nil {
+				return nil, err
+			}
+			max, err := strconv.Atoi(parts[3])
+			if err != nil {
+				return nil, err
+			}
+
+			return chunk.NewRabinMinMax(r, uint64(min), uint64(avg), uint64(max)), nil
+		default:
+			return nil, errors.New("incorrect format (expected 'rabin' 'rabin-[avg]' or 'rabin-[min]-[avg]-[max]'")
+		}
+	default:
+		return nil, fmt.Errorf("unrecognized chunker option: %s", chunker)
+	}
```

Comment:
this is pretty cool, i like being able to specify options. should it be on the chunker or separate? i.e.:

```
ipfs add -s=rabin-1000-10000
ipfs add -s=rabin --rabin-min=1000 --rabin-max=10000
```
less typing, but also more clear.

i do like being able to express it in one string, but dont like the non-self-description. it may be worth expressing all chunckers in one string:

```
ipfs add -s=rabin-min:1000-max:10000
```
not sure. maybe the `min:` and `max:` parts could be optional?

btw, we can probably put the "string -> splitter with options" conversion in the splitter package, and use it in other tools. easy place to plug in other splitters over time.

also, "splitter" or "chunker" ? (last chance to rename). i like both.